gittrigger/gittrigger.go

```diff
@@ -35,12 +35,36 @@ func (g *gitTrigger) getLatestCommit() error {
 	return nil
 }
 
-func (g gitTrigger) checkTrigger() bool {
+func (g *gitTrigger) GetRemote() (string, error) {
+	var out bytes.Buffer
+	cmd := exec.Command("git", "remote")
+	cmd.Stdout = &out
+	err := cmd.Run()
+
+	if err != nil {
+		return "", err
+	}
+	remote_name := strings.TrimSpace(out.String())
+
+	out.Reset()
+	cmd.Stdout = &out
+
+	cmd = exec.Command("git", "remote", "get-url", "--all", remote_name)
```

Comment:
This worries me a bunch, since this could return a url like `http://user:pass@examplegitserver/repo.git`, and imo that's far too sensitive/risky.

Reply:
Great point; forgot some people don't use ssh keys - I'll parse it, strip any creds.

Reply:
Fixed in 1b02355